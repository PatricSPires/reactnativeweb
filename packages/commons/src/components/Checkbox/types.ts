import * as React from 'react'
import { ThemeType } from '../../styles/types'

export type CheckboxProps = {
  onPress?: () => void
  children?: React.ReactNode
  checked?: boolean
  disabled?: boolean
  theme?: ThemeType
}
