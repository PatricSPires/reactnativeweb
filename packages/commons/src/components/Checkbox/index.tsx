import React from 'react'
import { StyleSheet, View } from 'react-native'
import {
  CheckBoxCheck,
  CheckBoxText,
  CheckBoxWrapper,
  Container,
} from './styles'
import { CheckboxProps } from './types'

const Checkbox = ({ children, checked, onPress, disabled }: CheckboxProps) => (
  <Container accessibilityRole="checkbox" disabled={disabled} onPress={onPress}>
    <View style={styles.wrapper}>
      <CheckBoxWrapper checked={checked} disabled={disabled}>
        <CheckBoxCheck checked={checked} disabled={disabled} />
      </CheckBoxWrapper>
      {children && <CheckBoxText disabled={disabled}>{children}</CheckBoxText>}
    </View>
  </Container>
)

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export default Checkbox
