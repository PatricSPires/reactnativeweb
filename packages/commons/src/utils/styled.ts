import styleSheetSerializer from 'jest-styled-components/src/styleSheetSerializer'

export function enableStyledComponentsSerializer() {
  expect.addSnapshotSerializer(styleSheetSerializer)
}
