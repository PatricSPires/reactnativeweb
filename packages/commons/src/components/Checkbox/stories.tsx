import React from 'react'
import { Story } from '@storybook/react/types-6-0'
import Checkbox from './index'
import { CheckboxProps } from './types'

export default {
  title: 'Formulário/Checkbox',
  component: Checkbox,
  parameters: {
    docs: {
      description: {
        component:
          'Este é o nosso componente de checkbox, que possui alguns estilos com base nas propriedades e estados.',
      },
    },
  },
  args: {
    children: 'Place your text here',
    checked: false,
    disabled: false,
  },
  argTypes: {
    onPress: {
      action: 'clicked',
      description: 'Muda o estado do componente para verdadeiro ou falso.',
    },
    children: {
      description: 'Texto que acompanha o checkbox. Não é obrigatório.',
    },
    checked: {
      description: 'Alterna o estado do componente entre verdadeiro e falso.',
    },
    disabled: {
      description:
        'Altera o estado do componente para desabilitado e não permite clique.',
    },
  },
}

export const Basic: Story<CheckboxProps> = (args) => <Checkbox {...args} />

export const Active = Basic.bind({})
Active.parameters = {
  docs: {
    description: {
      story:
        'Estado do componente quando está com a propriedade __checked__:true.',
    },
  },
}

Active.args = {
  checked: true,
}

export const Disabled = Basic.bind({})
Disabled.parameters = {
  docs: {
    description: {
      story:
        'Estado do componente quando está com a propriedade __disabled__:true.',
    },
  },
}

Disabled.args = {
  disabled: true,
  checked: false,
}

export const ActiveDisabled = Basic.bind({})
ActiveDisabled.parameters = {
  docs: {
    description: {
      story:
        'Estado do componente quando está com a propriedade __disabled__:true e __checked__:true.',
    },
  },
}

ActiveDisabled.args = {
  disabled: true,
  checked: true,
}
