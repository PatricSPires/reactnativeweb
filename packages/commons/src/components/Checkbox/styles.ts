import { Text, TouchableWithoutFeedback, View } from 'react-native'
import styled from 'styled-components/native'
import theme from '../../styles/theme'
import { CheckboxProps } from './types'

export const CheckBoxWrapper = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: ${theme.colors.high_contrast_border};
  border-radius: 2px;
  display: flex;
  align-items: center;
  justify-content: center;

  img {
    width: 15px;
    height: 15px;
  }

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === false) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }

    if (checked === false && disabled === true) {
      return `
        border-color: ${theme.colors.primary_subtle_border};
        background-color: transparent;
      `
    }
    return ``
  }}
`

export const CheckBoxCheck = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === true) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }
    return ``
  }}
`

export const CheckBoxText = styled(Text)`
  margin-left: 8px;
  color: ${(props: CheckboxProps) =>
    props.disabled
      ? theme.colors.primary_subtle_border
      : theme.colors.high_contrast_border};
`

export const Container = styled(TouchableWithoutFeedback)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`
