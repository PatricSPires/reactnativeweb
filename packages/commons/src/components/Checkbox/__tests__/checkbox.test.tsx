import { render } from '@testing-library/react-native'
import * as React from 'react'
import { enableStyledComponentsSerializer } from '../../../utils/styled'
import Checkbox from '../index'

beforeAll(() => {
  enableStyledComponentsSerializer()
})

describe('Checkbox Test', () => {
  it('should render the checkbox', () => {
    const { toJSON } = render(<Checkbox />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the checkbox with checked true', () => {
    const { toJSON } = render(<Checkbox checked />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the checkbox with checked false', () => {
    const { toJSON } = render(<Checkbox checked={false} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the checkbox with disabled true', () => {
    const { toJSON } = render(<Checkbox disabled />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the checkbox with disabled false', () => {
    const { toJSON } = render(<Checkbox disabled={false} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render the checkbox with children', () => {
    const { toJSON } = render(<Checkbox>Hello World</Checkbox>)

    expect(toJSON()).toMatchSnapshot()
  })
})
