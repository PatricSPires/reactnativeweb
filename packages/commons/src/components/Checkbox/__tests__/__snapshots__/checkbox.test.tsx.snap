// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Test should render the checkbox 1`] = `
.c2 {
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 15px;
  height: 15px;
}

.c3 {
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;
  background-size: 6px;
  background-repeat: no-repeat;
  background-position: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="checkbox"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
    />
  </div>
</div>
`;

exports[`Checkbox Test should render the checkbox with checked false 1`] = `
.c2 {
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 15px;
  height: 15px;
}

.c3 {
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;
  background-size: 6px;
  background-repeat: no-repeat;
  background-position: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="checkbox"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
    />
  </div>
</div>
`;

exports[`Checkbox Test should render the checkbox with checked true 1`] = `
.c2 {
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 15px;
  height: 15px;
}

.c3 {
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;
  background-size: 6px;
  background-repeat: no-repeat;
  background-position: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="checkbox"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
    />
  </div>
</div>
`;

exports[`Checkbox Test should render the checkbox with children 1`] = `
.c2 {
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 15px;
  height: 15px;
}

.c3 {
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;
  background-size: 6px;
  background-repeat: no-repeat;
  background-position: center;
}

.c4 {
  margin-left: 8px;
  color: #161616;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c5:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="checkbox"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
    />
  </div>
  <div
    className="c4 sc-dlnjPT c4 css-text-901oao"
    dir="auto"
  >
    Hello World
  </div>
</div>
`;

exports[`Checkbox Test should render the checkbox with disabled false 1`] = `
.c2 {
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 15px;
  height: 15px;
}

.c3 {
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;
  background-size: 6px;
  background-repeat: no-repeat;
  background-position: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c0:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c4:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-cursor-1loqt21 r-flexDirection-18u37iz r-justifyContent-1777fci r-touchAction-1otgn73"
  data-focusable={true}
  disabled={false}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="checkbox"
  tabIndex="0"
>
  <div
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={false}
  >
    <div
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={false}
    />
  </div>
</div>
`;

exports[`Checkbox Test should render the checkbox with disabled true 1`] = `
.c2 {
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: #161616;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 img {
  width: 15px;
  height: 15px;
}

.c3 {
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;
  background-size: 6px;
  background-repeat: no-repeat;
  background-position: center;
}

.c4:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
}

.c0:focus .c1 {
  box-shadow: 0px 0px 0px 1px #0043CE;
}

<div
  aria-disabled={true}
  className="c0 sc-hKFyIo c0 css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="checkbox"
>
  <div
    aria-disabled={true}
    className="c1 c2 c1 c2 css-view-1dbjc4n"
    disabled={true}
  >
    <div
      aria-disabled={true}
      className="c3 sc-gtssRu c3 css-view-1dbjc4n"
      disabled={true}
    />
  </div>
</div>
`;
